modules/library/src/main/scala/zio/elasticsearch/ElasticRequest.scala

```diff
@@ -205,11 +253,17 @@ object ElasticRequest {
     routing: Option[Routing]
   ) extends ElasticRequest[Option[Document], GetById]
 
-  private[elasticsearch] final case class GetByQueryRequest(
+  private[elasticsearch] final case class SearchRequest(
     index: IndexName,
-    query: ElasticQuery[_],
+    query: Option[ElasticQuery[_]],
+    aggregation: Option[ElasticAggregation],
     routing: Option[Routing]
-  ) extends ElasticRequest[ElasticQueryResponse, GetByQuery]
+  ) extends ElasticRequest[ElasticQueryResponse, Search]
+
+  private[elasticsearch] final case class SearchResponse[A](
```

Comment:
Let's add the methods we were talking about... `getAggregation` or `aggregation(name: String)`/ Also, does it make sense to rename it to `ElasticResponse`?

Reply:
I guess we should add `items` or `data` as well.

Reply:
We already have `ElasticGetResponse`, etc.. So i guess it can be renamed to `ElasticSearchResponse`?